homeassistant/components/shopping_list/__init__.py

```diff
@@ -375,18 +420,38 @@ async def websocket_handle_add(
     connection: websocket_api.ActiveConnection,
     msg: dict[str, Any],
 ) -> None:
-    """Handle add item to shopping_list."""
+    """Handle adding item to shopping_list."""
     item = await hass.data[DOMAIN].async_add(msg["name"], connection.context(msg))
     connection.send_message(websocket_api.result_message(msg["id"], item))
 
 
+@websocket_api.async_response
+async def websocket_handle_remove(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """Handle removing shopping_list item."""
+    msg_id = msg.pop("id")
+    item_id = msg.pop("item_id")
+    msg.pop("type")
+
+    try:
+        item = await hass.data[DOMAIN].async_remove(item_id, connection.context(msg))
+        connection.send_message(websocket_api.result_message(msg_id, item))
```

Comment:
Sending the command should not raise `KeyError`, right? Please only wrap the line that we expect to raise in the try... except block.